packages/ses/src/module-instance.js

```diff
@@ -444,11 +445,16 @@ export const makeModuleInstance = (
     activate();
   }
 
-  let optFunctor = compartmentEvaluate(compartmentFields, functorSource, {
-    globalObject: compartment.globalThis,
-    transforms: __shimTransforms__,
-    __moduleShimLexicals__: moduleLexicals,
-  });
+  let optFunctor;
+  if (__precompiledFunctor__ !== undefined) {
+    optFunctor = __precompiledFunctor__;
```

Comment:
Our theory is that this is safe because guest code can only create confined functors and host code is necessarily responsible for participating in the enforcement of confinement. cc @erights because this is an area where we’re likely to miss something.

The other half of the theory is that this violates Hyrum’s law but only in a way that we’re already breaking Hyrum’s law. Code can be written to depend on this behavior that will necessarily break on a standard implementation of Compartment sans shim. That doesn’t preclude the possibility of the shim standing atop a genuine implementation and providing a backward compatibility layer, though.

Reply:
And let’s capture this reasoning in a comment for posterity.

Reply:
I suppose the main thing that needs to be evaluated is whether the argument passed to the functor reveals anything the code running in the compartment may not have had access to otherwise?

Reply:
I believe that’s the nature of the question.

This path skips `evaluate` so I do not believe we have concerns about lifting text to behavior safely. The functor receives an `imports` object from the module system which grants the module the ability to get and set values of its environment. If that provides excess authority, it does so in a way that’s already achievable by fabricating a static module record duck type for a child compartment.